server/asset/eth/eth.go

```diff
@@ -238,6 +238,30 @@ func (eth *Backend) FeeRate(ctx context.Context) (uint64, error) {
 	return dexeth.WeiToGweiUint64(bigGP)
 }
 
+// SupportsDynamicTxFee returns true if the tx fee for this asset adjusts based
+// on market conditions
+func (*Backend) SupportsDynamicTxFee() bool {
+	return true
+}
+
+// ValidateFeeRate checks that the transaction fees used to initiate the
+// contract are sufficient. For most assets only the contract.FeeRate() cannot
+// be less than reqFeeRate, but for Eth, the gasTipCap must also be checked.
+func (eth *Backend) ValidateFeeRate(contract *asset.Contract, reqFeeRate uint64) bool {
+	coin := contract.Coin
+	sc, ok := coin.(*swapCoin)
+	if !ok {
+		eth.log.Error("eth contract coin type must be a swapCoin but got %T", sc)
+		return false
+	}
+
+	if sc.dynamicTx && sc.gasTipCap < dexeth.MinGasTipCap {
```

Comment:
This seems good, in that `ValidateFeeRate` will facilitate a more sophisticated `gasTipCap` check than just comparing `MinGasTipCap`, such as something based on a running average of recent blocks tips, but as discussed in planning this, I believe a constant will be quite adequate.

Reply:
I thought this would be adequate also. I generally 1 gwei is enough for the tip cap. If 2 gwei isn't enough it would be some kind of extreme network conditions, and even then the client still checks the suggested tip cap. It's just that the server doesn't do more complex verification.